modules/loggers/mod_log_config.c

```diff
@@ -1902,6 +2239,51 @@ static int log_pre_config(apr_pool_t *p, apr_pool_t *plog, apr_pool_t *ptemp)
         log_pfn_register(p, "x", log_ssl_var, 0);
     }
 
+    /*
+     * should align with tomcat attribute names:
+     * https://github.com/apache/tomcat/blob/9.0.x/java/org/apache/catalina/valves/JsonAccessLogValve.java#L78
+     * also interessting is fluentd resp. fluent-bit:
+     * - https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/parser_apache2.rb#L72
+     */
+    if (log_pfn_register) {
+        json_register_attribute(p, "a", "remoteAddr");
```

Comment:
Rather than `json_register_attribute()`, it would be cleaner to implement an extended `log_pfn_register()` function, something like `log_pfn_register_ex()`, that takes the extra long form of the name.

`log_pfn_register()` can then be kept for backwards compatibility, with the long form name being NULL. Any module that doesn't provide a long form parameter is just ignored by the json formatter.

This will be safe to backport to v2.4.

Reply:
I added log_pfn_register_ex()